<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>481</width>
    <height>782</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>设置</string>
  </property>
  <layout class="QFormLayout" name="formLayout">
   <item row="0" column="0" colspan="2">
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>标定方法:</string>
     </property>
     <property name="checkable">
      <bool>false</bool>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QRadioButton" name="rbtn_dlt">
        <property name="text">
         <string>DLT</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="0" colspan="2">
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>初始化方法:</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QRadioButton" name="rbtn_manual">
        <property name="text">
         <string>手工调整</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QRadioButton" name="rbtn_epnp">
        <property name="text">
         <string>EPnP约束(需要2D与3D关键点一致对应,且N&gt;=4)</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QRadioButton" name="rbtn_pso">
        <property name="text">
         <string>PSO (粒子群算法)</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QRadioButton" name="rbtn_none">
        <property name="text">
         <string>不初始化</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="0" colspan="2">
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>物体1-选点方式:</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QRadioButton" name="rbtn_ellipse">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>第2个点往后满足椭圆约束</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="rbtn_correspond">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="focusPolicy">
         <enum>Qt::StrongFocus</enum>
        </property>
        <property name="text">
         <string>与3D关键点一致</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
        <property name="autoExclusive">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="4" column="0" colspan="2">
    <widget class="QGroupBox" name="groupBox_4">
     <property name="title">
      <string>超参数:</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QGroupBox" name="groupbox_hperparams_adam">
        <property name="title">
         <string>Adam</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="spinbox_alpha_adam">
           <property name="decimals">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QDoubleSpinBox" name="spinbox_beta2_adam">
           <property name="decimals">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QDoubleSpinBox" name="spinbox_beta1_adam">
           <property name="decimals">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_niters_adam">
           <property name="text">
            <string>迭代次数:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_alpha_adam">
           <property name="text">
            <string>初始学习率:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_beta1_adam">
           <property name="text">
            <string>衰减参数1:</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_beta2_adam">
           <property name="text">
            <string>衰减参数2:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="spinbox_niters_adam">
           <property name="maximum">
            <number>100000</number>
           </property>
           <property name="singleStep">
            <number>100</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupbox_hperparams_lm">
        <property name="title">
         <string>Levenberg Marquardt(LM)</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="2" column="1">
          <widget class="QDoubleSpinBox" name="doubleSpinBox_5">
           <property name="decimals">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="spinbox_alpha_lm">
           <property name="decimals">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QDoubleSpinBox" name="doubleSpinBox_6">
           <property name="decimals">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_niters_lm">
           <property name="text">
            <string>迭代次数:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_alpha_lm">
           <property name="text">
            <string>初始学习率:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>TBD</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_8">
           <property name="text">
            <string>TBD</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="spinbox_niters_lm">
           <property name="maximum">
            <number>100000</number>
           </property>
           <property name="singleStep">
            <number>100</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupbox_hperparams_pso">
        <property name="title">
         <string>Particle Swarm Optimization(PSO):</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
        <layout class="QGridLayout" name="gridLayout_4">
         <item row="3" column="0">
          <widget class="QLabel" name="label_c1_pso">
           <property name="text">
            <string>学习因子1:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_niters_pso">
           <property name="text">
            <string>迭代次数:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_npops_pso">
           <property name="text">
            <string>粒子数量:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_w_pso">
           <property name="text">
            <string>惯性因子:</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_c2_pso">
           <property name="text">
            <string>学习因子2:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QDoubleSpinBox" name="spinbox_w_pso">
           <property name="decimals">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QDoubleSpinBox" name="spinbox_c1_pso">
           <property name="decimals">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QDoubleSpinBox" name="spinbox_c2_pso">
           <property name="decimals">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="spinbox_niters_pso">
           <property name="maximum">
            <number>100000</number>
           </property>
           <property name="singleStep">
            <number>100</number>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="spinbox_npops_pso"/>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="8" column="1">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="inputMethodHints">
      <set>Qt::ImhNone</set>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
     <property name="centerButtons">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="3" column="0" colspan="2">
    <widget class="QGroupBox" name="groupBox_8">
     <property name="title">
      <string>优化器:</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QRadioButton" name="rbtn_adam">
        <property name="text">
         <string>Adam</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="rbtn_lm">
        <property name="text">
         <string>LM</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>Dialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>Dialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rbtn_adam</sender>
   <signal>toggled(bool)</signal>
   <receiver>groupbox_hperparams_adam</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>128</x>
     <y>222</y>
    </hint>
    <hint type="destinationlabel">
     <x>240</x>
     <y>333</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rbtn_pso</sender>
   <signal>toggled(bool)</signal>
   <receiver>groupbox_hperparams_pso</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>377</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>240</x>
     <y>618</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rbtn_lm</sender>
   <signal>toggled(bool)</signal>
   <receiver>groupbox_hperparams_lm</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>352</x>
     <y>222</y>
    </hint>
    <hint type="destinationlabel">
     <x>240</x>
     <y>469</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>groupbox_hperparams_adam</sender>
   <signal>clicked(bool)</signal>
   <receiver>rbtn_adam</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>240</x>
     <y>333</y>
    </hint>
    <hint type="destinationlabel">
     <x>128</x>
     <y>222</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>groupbox_hperparams_lm</sender>
   <signal>clicked(bool)</signal>
   <receiver>rbtn_lm</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>240</x>
     <y>469</y>
    </hint>
    <hint type="destinationlabel">
     <x>352</x>
     <y>222</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>groupbox_hperparams_pso</sender>
   <signal>clicked(bool)</signal>
   <receiver>rbtn_pso</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>240</x>
     <y>618</y>
    </hint>
    <hint type="destinationlabel">
     <x>377</x>
     <y>92</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
